api/km-storage.js

```diff
@@ -185,13 +187,38 @@ class KmAttachmentsApi extends ApiBase
     super(options);
   }
 
-  async uploadTempFile(body, params)  {
-    // const headers = {'Content-Type': undefined };
-    const data =  await useAPIFetch('/api/v2015/temporary-files',   { method:'post', body, params })
-    //required by ckeditor
-    data.urls = data.urls || [data.url];
+  async uploadTempFile(identifier, file, fileName, options)  {
     
-    return data;
+    
+    const { headers, timeout, contentType, onUploadProgress, onDownloadProgress }= { ...(options||{}) };
+
+    const apiConfig = {
+      headers: headers ||{},
+      timeout: timeout || 60 * 60 * 1000,
+      onUploadProgress, 
+      onDownloadProgress
+    };
```

Comment:
yes the header must be a mistake, I am just passing the filename and then using the return object  from temp api to set the content-type. but i think it will secure to read and pass the content-type from the html object to temp api to get slot